### PR TITLE
feat(game-client): add interactive translit output tooltips with segment mapping

### PR DESCRIPTION
**Pull request by rocescoca**
## Description

This PR upgrades the transliteration output surface from a read-only textarea to a tokenized, interactive view that shows source-script segments in tooltips on hover/focus. It introduces shared tooltip styles, segment-splitting utilities, and a dedicated `TranslitOutput` component, then wires them into `TranslitClient` with scroll sync preserved between input/output panes.

It also improves output readability and layout behavior by adding adaptive text sizing for longer content, better overflow handling in split panels, and Storybook coverage plus unit tests for segment generation edge cases (punctuation, repeated whitespace, and line breaks).

---

## Linked Issues

Closes #

---

## Type

- [x] `feat` - New feature
- [ ] `chore` - Infrastructure, setup tasks, non-feature work
- [ ] `fix` - Bug fix
- [ ] `docs` - Documentation changes
- [x] `test` - Testing-related changes

---

## Scope

- [x] `game-client` - Game client application
- [ ] `backoffice-client` - Backoffice client application
- [ ] `ui` - UI package
- [x] `storybook` - Storybook tool
- [ ] `e2e` - E2E testing
- [ ] `global` - Shared packages or general repository tasks

---

## Screenshots (Optional)

N/A

---

## Preview Links (Optional)

N/A

---

## Testing Notes

- [ ] Tests pass locally
- [ ] Linting passes

- Added unit tests for `getTranslitOutputSegments` behavior across whitespace/newline and punctuation cases.
- Added Storybook interaction stories validating tooltip visibility on hover and keyboard focus.
- Verify in app: input/output remain scroll-synced and output tokens reveal matching source segments.

---

## Documentation Changes (if applicable)

- [ ] Added or updated documentation files
- [ ] Updated cross-references in other docs (if files were renamed/moved)
- [ ] Verified all internal links still work
- [x] N/A - No documentation changes

## Summary by CodeRabbit

* **New Features**
  * Added TranslitOutput component with interactive tooltips displaying transliteration source text on hover or focus
  * Support for touch/tap interactions with automatic tooltip management on mobile devices
  * Compact text mode that dynamically adjusts typography size based on content height
  * Scroll synchronization between input and output surfaces

* **Tests**
  * Added test coverage for output segment functionality and segmentation logic

* **Documentation**
  * Added Storybook stories documenting TranslitOutput component behavior and interactions

* **Style**
  * Consolidated tooltip styling into shared module for consistency

### CODE REVIEW

File: apps/game-client/src/ui/screens/translit/translit-client.tsx [hunk 1]
```diff
@@ -57,13 +70,16 @@ export function TranslitClient({ library }: Readonly<{ library: Library }>) {
...
   const [isCopySuccess, setIsCopySuccess] = useState(false);
+  const [isCompactText, setIsCompactText] = useState(false);
   /** Hides copy tooltip as soon as the button is pressed (before async clipboard), so toast is not doubled with an open hover tooltip */
   const [suppressCopyTooltip, setSuppressCopyTooltip] = useState(false);
   const [direction, setDirection] = useState<'georgian-to-latin' | 'latin-to-georgian'>(
     'georgian-to-latin',
   );
   const inputRef = useRef<HTMLTextAreaElement>(null);
-  const outputRef = useRef<HTMLTextAreaElement>(null);
+  const outputRef = useRef<HTMLDivElement>(null);
```
[the remark this answers is not in the document]
Comment (rocescoca) [reply apps/game-client/src/ui/screens/translit/translit-client.tsx]: Good catch. I updated both refs to use HTMLElement so the types match the actual <section>-backed output surface.

File: apps/game-client/src/ui/screens/translit/translit-client.tsx [hunk 2]
```diff
@@ -203,9 +219,55 @@ export function TranslitClient({ library }: Readonly<{ library: Library }>) {
...
+      updateTextSize();
+    });
+
+    resizeObserver.observe(inputRef.current);
+    resizeObserver.observe(outputRef.current);
+
+    return () => {
+      resizeObserver.disconnect();
+    };
+  }, [input, output, direction]);
```
[the remark this answers is not in the document]
Comment (rocescoca) [reply apps/game-client/src/ui/screens/translit/translit-client.tsx]: Agreed. I split the measurement logic from the observer lifecycle so the ResizeObserver is stable and only the measurement runs on content changes.

File: apps/game-client/src/ui/screens/translit/translit-client.tsx [hunk 3]
```diff
@@ -223,6 +285,12 @@ export function TranslitClient({ library }: Readonly<{ library: Library }>) {
 
   const inputLang = direction === 'georgian-to-latin' ? 'ka-GE' : 'ka-Latn';
   const outputLang = direction === 'georgian-to-latin' ? 'ka-Latn' : 'ka-GE';
+  const outputSegments = getTranslitOutputSegments(input, output);
```
[the remark this answers is not in the document]
Comment (rocescoca) [reply apps/game-client/src/ui/screens/translit/translit-client.tsx:288]: I’m leaving this one as-is for now. The repo’s frontend guidance avoids adding useMemo by default, and I’d rather profile before adding memoization here.

File: apps/game-client/src/ui/screens/translit/translit-output-segments.test.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+  it('preserves line breaks inside text runs when splitting only on spaces', () => {
+    expect(getTranslitOutputSegments('აბ\nგდ ე', 'ab\ngd e')).toEqual([
+      { id: '0:აბ:ab', sourceText: 'აბ', outputText: 'ab', isWhitespace: false },
+      { id: '1:\n:\n', sourceText: '\n', outputText: '\n', isWhitespace: true },
+      { id: '2:გდ:gd', sourceText: 'გდ', outputText: 'gd', isWhitespace: false },
+      { id: '3: : ', sourceText: ' ', outputText: ' ', isWhitespace: true },
+      { id: '4:ე:e', sourceText: 'ე', outputText: 'e', isWhitespace: false },
+    ]);
+  });
```
[the remark this answers is not in the document]
Comment (rocescoca) [reply apps/game-client/src/ui/screens/translit/translit-output-segments.test.ts]: Agreed. I renamed the test so it reflects the current tokenizer behavior where line breaks are emitted as their own whitespace segments.

File: apps/game-client/src/ui/screens/translit/translit-output-segments.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
+export type TranslitOutputSegment = {
+  id: string;
+  sourceText: string;
+  outputText: string;
+  isWhitespace: boolean;
+};
+
+const segmentPattern = /[ \n]+|[^ \n]+/g;
+const whitespaceRunPattern = /^[ \n]+$/;
```
[the remark this answers is not in the document]
Comment (rocescoca) [reply apps/game-client/src/ui/screens/translit/translit-output-segments.ts]: Agreed. I broadened the tokenizer to use \\s / \\S so pasted tabs, CR, NBSP, and other standard whitespace are treated as separators too.

File: apps/game-client/src/ui/screens/translit/translit-output.tsx [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+                  <Tooltip.Arrow className={translitTooltipArrowClassName}>
+                    <div className={translitTooltipArrowInnerClassName} />
+                  </Tooltip.Arrow>
+                  <span>{segment.sourceText}</span>
+                </Tooltip.Popup>
+              </Tooltip.Positioner>
+            </Tooltip.Portal>
+          </Tooltip.Root>
+        );
+      })}
```
[the remark this answers is not in the document]
Comment (rocescoca) [reply apps/game-client/src/ui/screens/translit/translit-output.tsx:192]: I’m not changing this one. The tokens already render under the existing Tooltip.Provider higher in the translit screen, so the recommendation is already satisfied without adding another provider layer.
